xtra-libp2p/src/endpoint.rs

```diff
@@ -210,15 +242,18 @@ impl Endpoint {
             listen_addresses: HashSet::default(),
             inflight_connections: HashSet::default(),
             connection_timeout,
+            subscribers,
         }
     }
 
-    fn drop_connection(&mut self, peer: &PeerId) {
+    async fn drop_connection(&mut self, peer: &PeerId) {
         let (mut control, tasks) = match self.controls.remove(peer) {
             None => return,
             Some(control) => control,
         };
 
+        self.notify_connection_dropped(*peer).await;
```

Comment:
> I'm not sure how making this async could affect the overall performance of the `Endpoint`.

good question. thanks for pointing out that we could do it in the task below - I opted for reverting this function to not be async, and notifying after the drop (this is actually more correct, because don't advertise something as dropped anymore before it actually got dropped).
Thanks!

Reply:
edit: turns out that we can't notify it from the task, because it needs access to `&self` :/ I opted for moving it *after* the drop for now. if it will become problematic, we'll revisit the pattern.

Reply:
The point of an async dispatch is that it is non-blocking, i.e. you dispatch and don't wait for any response. If this does not work as expected we generally have a problem.